nservicebus/azure-service-bus/performance-tuning_body_asb_[7,).partial.md

```diff
@@ -0,0 +1,61 @@
+## Tuning send performance outside the scope of a handler
+
+Only settings at the ASB SDK level impact this scenario, more specifically
+
+  * `MessagingFactories().BatchFlushInterval()`: Controls the time that the ASB SDK batches client side requests before sending them out. A value of TimeSpan.Zero turns this feature off. Values between 100 and 500 milliseconds appear to work best.
+  * `MessagingFactories().NumberOfMessagingFactoriesPerNamespace()`: Each factory manages a tcp connection to a front end node in the broker, this connection has throughput limits. Opening more factories significantly improves send performance. Values between 2 and 8 seem to work best for sending.
+  * `NumberOfClientsPerEntity()`: Keep this value equal to the number of factories, this ensures there is 1 send client per factory.
+
+The way the send api is invoked matters a lot as well. Using `await` on each `Send()` operation will make the ASB SDK wait until the send operation completes and therefore makes client side batching useless. It's better to maintain a list of send tasks and await them all together instead.
+
+Slow:
+
+Snippet: asb-slow-send
+
+Fast:
+
+Snippet: asb-fast-send
+
+## Tuning receive performance
+
+Tuning receive performance depends on quite a few things that vary per scenario, but the following settings bring improvements for all scenarios.
+
+  * `MessagingFactories().NumberOfMessagingFactoriesPerNamespace()`: Each factory manages a tcp connection to a front end node in the broker, this connection has throughput limits. Opening more factories significantly improves receive performance. Values between 16 and 40 seem to work best for receiving.
+  * `NumberOfClientsPerEntity()`: Keep this value equal to the number of factories, this ensures there is 1 receive client per factory.
+
+Other settings that impact performance a lot, but their values differ quite a bit per scenario, are:
+
+  * `LimitMessageProcessingConcurrencyTo()`: This is the global concurrency limit across all receive entities. Ensure this value matches the per receiver desired concurrency for the scenario, multiplied by the number of receivers.
```

Comment:
@SeanFeldman what do you mean exactly? The ratio is really different per transaction mode, can't generalize it